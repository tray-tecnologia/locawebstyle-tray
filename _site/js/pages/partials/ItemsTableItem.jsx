import React from 'react';

class ItemsTableItem extends React.Component {
    render() {
        if (!this.props.item) {
            throw new Error('Prop "item" is missing!');
        }

        return (
            <tr>
                <td>
                    <input type="checkbox" data-checkbox-toggle="items-table" />
                </td>
                <td>{this.props.item.id}</td>
                <td>{this.props.item.status ? 'Ativo' : 'Inativo'}</td>
                <td>
                    <a className="ls-ico-external" href={this.props.item.url}>{this.props.item.name}</a>
                </td>
                <td>{this.props.item.category}</td>
                <td>{this.props.item.price}</td>
                <td>{this.props.item.stock}</td>
                <td className="ls-txt-right">
                    <a href="#" className="ls-btn-default ls-display-none">Detalhes</a>
                    <div data-ls-module="dropdown" className="ls-dropdown">
                        <button className="ls-btn"></button>
                        <ul className="ls-dropdown-nav">
                            <li><a href="#">Editar</a></li>
                            <li><a href="#">Duplicar</a></li>
                            <li><a className="ls-text-danger" href="#">Excluir</a></li>
                        </ul>
                    </div>
                </td>
            </tr>
        );
    }
};

export default ItemsTableItem;