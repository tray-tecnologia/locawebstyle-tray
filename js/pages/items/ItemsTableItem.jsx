import React from 'react';

class ItemsTableItem extends React.Component {
    render() {
        if (!this.props.item) {
            throw new Error('Prop "item" is missing!');
        }

        return (
            <tr>
                <td>
                    <input checked={this.props.item.selected} type="checkbox" data-item-id={this.props.item.id} data-checkbox-toggle="items-table" onChange={this.props.onSelectItem} />
                </td>
                <td>{this.props.item.id}</td>
                <td>
                    <a className="ls-ico-external" href={this.props.item.url}>{this.props.item.name}</a>
                </td>
                <td>{this.props.item.category}</td>
                <td>{this.props.item.price}</td>
                <td>{this.props.item.stock}</td>
                <td className={'ls-txt-center ls-width-100 ' + (this.props.item.status ? 'ls-background-success': 'ls-background-danger')}>
                    {this.props.item.status ? 'Ativo' : 'Inativo'}
                </td>
                <td className="ls-txt-right">
                    <a href="#" className="ls-btn-default ls-display-none">Detalhes</a>
                    <button className="ls-btn">
                        <span className="ls-ico-list"></span>
                    </button>
                </td>
            </tr>
        );
    }
};

export default ItemsTableItem;