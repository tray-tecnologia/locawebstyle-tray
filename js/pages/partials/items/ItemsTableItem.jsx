import React from 'react';

class ItemsTableItem extends React.Component {
    render() {
        if (!this.props.item) {
            throw new Error('Prop "item" is missing!');
        }

        return (
            <tr>
                <td>
                    <input type="checkbox" data-checkbox-toggle="items-table" />
                </td>
                <td>{this.props.item.id}</td>
                <td>{this.props.item.status ? 'Ativo' : 'Inativo'}</td>
                <td>
                    <a className="ls-ico-external" href={this.props.item.url}>{this.props.item.name}</a>
                </td>
                <td>{this.props.item.category}</td>
                <td>{this.props.item.price}</td>
                <td>{this.props.item.stock}</td>
                <td className="ls-txt-right">
                    <a href="#" className="ls-btn-default ls-display-none">Detalhes</a>
                    <button className="ls-btn">
                        <span className="ls-ico-list"></span>
                    </button>
                </td>
            </tr>
        );
    }
};

export default ItemsTableItem;