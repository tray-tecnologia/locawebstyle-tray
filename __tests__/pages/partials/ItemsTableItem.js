import React from 'react';
import renderer from 'react-test-renderer';

import ItemsTableItem from '../../../js/pages/partials/ItemsTableItem.jsx';

describe('TableItem', () => {
    it('show a row with active item', () => {
        const item = {
            id: 1,
            status: true,
            url: '/items/1',
            name: 'Item 1',
            category: 'Items',
            price: 9.34,
            stock: 35
        };

        const itemsTableItemComp = renderer.create(
            <ItemsTableItem item={item} />
        );

        expect(itemsTableItemComp.toJSON()).toMatchSnapshot();
    });

    it('show a row with inactive item', () => {
        const item = {
            id: 1,
            status: false,
            url: '/items/1',
            name: 'Item 1',
            category: 'Items',
            price: 9.34,
            stock: 35
        };

        const itemsTableItemComp = renderer.create(
            <ItemsTableItem item={item} />
        );

        expect(itemsTableItemComp.toJSON()).toMatchSnapshot();
    });

    it('show a row with no item', () => {
        expect(() => {
            renderer.create(
                <ItemsTableItem />
            )
        }).toThrowError(/missing/);
    });
});
