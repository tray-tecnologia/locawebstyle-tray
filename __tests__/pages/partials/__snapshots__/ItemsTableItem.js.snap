exports[`TableItem show a row with active item 1`] = `
<tr>
  <td>
    <input
      data-checkbox-toggle="items-table"
      type="checkbox" />
  </td>
  <td>
    1
  </td>
  <td>
    Ativo
  </td>
  <td>
    <a
      className="ls-ico-external"
      href="/items/1">
      Item 1
    </a>
  </td>
  <td>
    Items
  </td>
  <td>
    9.34
  </td>
  <td>
    35
  </td>
  <td
    className="ls-txt-right">
    <a
      className="ls-btn-default ls-display-none"
      href="#">
      Detalhes
    </a>
    <button
      className="ls-btn">
      <span
        className="ls-ico-list" />
    </button>
  </td>
</tr>
`;

exports[`TableItem show a row with inactive item 1`] = `
<tr>
  <td>
    <input
      data-checkbox-toggle="items-table"
      type="checkbox" />
  </td>
  <td>
    1
  </td>
  <td>
    Inativo
  </td>
  <td>
    <a
      className="ls-ico-external"
      href="/items/1">
      Item 1
    </a>
  </td>
  <td>
    Items
  </td>
  <td>
    9.34
  </td>
  <td>
    35
  </td>
  <td
    className="ls-txt-right">
    <a
      className="ls-btn-default ls-display-none"
      href="#">
      Detalhes
    </a>
    <button
      className="ls-btn">
      <span
        className="ls-ico-list" />
    </button>
  </td>
</tr>
`;
